code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -52,22 +52,22 @@
 		return FALSE
 
 	if(istype(A, /obj/item/storage/pill_bottle) && is_open_container()) //this should only run if its a pillbottle
+		var/obj/item/storage/pill_bottle/bottle = A
 		if(reagents.total_volume >= volume)
 			to_chat(user, span_warning("[src] is full."))
 			return  //early returning if its full
 
-		if(!length(A.contents))
+		if(!length(bottle.contents))
 			return //early returning if its empty
-		var/obj/item/pill = A.contents[1]
+		var/obj/item/pill = bottle.contents[1]
 
 		if((pill.reagents.total_volume + reagents.total_volume) > volume)
 			to_chat(user, span_warning("[src] cannot hold that much more."))
 			return // so it doesnt let people have hypos more filled than their volume
 		pill.reagents.trans_to(src, pill.reagents.total_volume)
 
-		to_chat(user, span_notice("You dissolve [pill] from [A] in [src]."))
-		A.contents -= pill
-		qdel(pill)
+		to_chat(user, span_notice("You dissolve [pill] from [bottle] in [src]."))
+		bottle.remove_from_storage(pill,user,user)
```

Comment:
No sense in moving the pill into the mob to hang out forever, just send it to nullspace and delete it.